rust/index/src/fulltext/types.rs

```diff
@@ -92,60 +93,71 @@ impl<'me> FullTextIndexWriter<'me> {
+        for token in tokens {
+            match uncommitted_frequencies.get(&token.text) {
+                Some(_) => return Ok(()),
+                None => {
+                    let frequency = match &self.full_text_index_reader {
+                        // Readers are uninitialized until the first compaction finishes
+                        // so there is a case when this is none hence not an error.
+                        None => 0,
+                        Some(reader) => {
+                            match reader.get_frequencies_for_token(token.text.as_str()).await {
+                                Ok(frequency) => frequency,
+                                // New token so start with frequency of 0.
+                                Err(_) => 0,
+                            }
+                        }
+                    };
+                    uncommitted_frequencies
+                        .insert(token.text.clone(), (frequency as i32, frequency as i32));
+                }
             }
         }
+
+        drop(uncommitted_frequencies);
+
         let mut uncommitted_postings = self.uncommitted_postings.lock().await;
-        match uncommitted_postings.positional_postings.get(token) {
-            Some(_) => {
-                // This should never happen -- if uncommitted has the token, then
-                // uncommitted_frequencies should have had it as well.
-                tracing::error!(
-                    "Error populating frequencies and posting lists from previous version"
-                );
-                return Err(FullTextIndexError::InvariantViolation);
-            }
-            None => {
-                let mut builder = PositionalPostingListBuilder::new();
-                let results = match &self.full_text_index_reader {
-                    // Readers are uninitialized until the first compaction finishes
-                    // so there is a case when this is none hence not an error.
-                    None => vec![],
-                    Some(reader) => match reader.get_all_results_for_token(token).await {
-                        Ok(results) => results,
-                        // New token so start with empty postings list.
-                        Err(_) => vec![],
-                    },
-                };
-                for (doc_id, positions) in results {
-                    let res = builder.add_doc_id_and_positions(doc_id as i32, positions);
-                    match res {
-                        Ok(_) => {}
-                        Err(e) => {
-                            return Err(FullTextIndexError::PositionalPostingListError(e));
+        for token in tokens {
+            match uncommitted_postings.positional_postings.get(&token.text) {
+                Some(_) => {
+                    // This should never happen -- if uncommitted has the token, then
+                    // uncommitted_frequencies should have had it as well.
+                    tracing::error!(
```

Comment:
Likewise this also seems incorrect. Previously we were relying on returning early but with continue in the loop above it will no longer be the case